### PR TITLE
Feature/update ghwfs yaml

### PR DESCRIPTION
**Pull request by Chargothrond**
Updated following mostly the latest example on https://github.com/r-lib/actions/tree/master/examples#standard-ci-workflow.

Align with the corresponding changes in https://github.com/miraisolutions/techguides/pull/27

### CODE REVIEW

File: .github/workflows/ci-cd.yml
Comment (riccardoporreca) [inline .github/workflows/ci-cd.yml:15]:
```suggestion
      # we keep a matrix for convenience, but we would typically just run on one
      # single OS and R version, aligned with the target deployment environment
```

**Review by riccardoporreca — CHANGES_REQUESTED** [review]
LGTM, just a re-wording about the single (OS, R) in the context of Shiny CI/CD

File: .github/workflows/ci-cd.yml
Comment (fvitalini) [reply .github/workflows/ci-cd.yml:15]: maybe it does not fit here as a comment, but it is not clear to me what a matrix implies and why is it desirable. Maybe worth adding a comment about this on the techguides

File: .github/workflows/ci-cd.yml
Comment (fvitalini) [inline .github/workflows/ci-cd.yml:63]: Can we add a comment about this line?

**Review by fvitalini — APPROVED** [review]

File: .github/workflows/ci-cd.yml
Comment (Chargothrond) [reply .github/workflows/ci-cd.yml:63]: We could, but then we should probably put a lot more comments in other sections, because this is not too relevant and anyone with some R background should be able to google this in 1-5 minutes (it basically just makes the github actions logs of the R CMD check step more colorful and prettier).

Personally, I don't think the yaml needs more comments, but it might be worth it since this repo is meant as an example to illustrate this.

So first we should decide in general if we want to put quite a few more comments into this file or not, opinions? @riccardoporreca @fvitalini @Guido (why can't I @ him here?)

Comment (fvitalini) [reply .github/workflows/ci-cd.yml:63]: I agree to have more comments. I think you cannot add Guido because he is not a contributor

**Comment by Chargothrond** [comment]
@fvitalini @riccardoporreca I added two comments on the file, not sure if anyone wants to add more, but I would really like to merge this PR soon, so please let me know what you think and if we should still add more comments or rephrase some (or in case you spot something else to be improved).

File: .github/workflows/ci-cd.yml
Comment (fvitalini) [inline .github/workflows/ci-cd.yml:20]: "Define a package manager for RStudio and expose the Github PAT for access"

If you say "we need to ..." then you need to tell me why

**Review by fvitalini — APPROVED** [review]
just some rewording for the comments

File: .github/workflows/ci-cd.yml
Comment (Chargothrond) [reply .github/workflows/ci-cd.yml]: updated

File: .github/workflows/ci-cd.yml
Comment (Chargothrond) [reply .github/workflows/ci-cd.yml]: updated, @riccardoporreca  please let me know in case there is something inaccurate with the updated comment

File: .github/workflows/ci-cd.yml
Comment (riccardoporreca) [inline .github/workflows/ci-cd.yml:66]: I would put this as an end-of-line comment not to give it too much importance, and simplify to tell what it does, not how ("option for ...")
```suggestion
          options(crayon.enabled = TRUE) # enable colorful R CMD check output
```

File: .github/workflows/ci-cd.yml
Comment (riccardoporreca) [reply .github/workflows/ci-cd.yml]: Doing this as part of the review comments

**Review by riccardoporreca — CHANGES_REQUESTED** [review]
See the proposed refinements about comments

**Comment by riccardoporreca** [comment]
We may also want to align some minor updates done in https://github.com/miraisolutions/techguides/pull/27 (esp. the deployment on `main` or `master`)

**Comment by Chargothrond** [comment]
@riccardoporreca I aligned the files as I understood and made sense to me.
Note that we currently get quite a different result, as calling `usethis::use_github_action(url = "https://mirai-solutions.ch/techguides/shiny-ci-cd/actions/ci-cd.yml")` actually evaluates things like `${{ matrix.config.os }}`. Not sure if you intended to revert these things on this repo, the files should otherwise now be perfectly aligned.

**Comment by riccardoporreca** [comment]
> Note that we currently get quite a different result, as calling `usethis::use_github_action(url = "https://mirai-solutions.ch/techguides/shiny-ci-cd/actions/ci-cd.yml")` actually evaluates things like `${{ matrix.config.os }}`

We want this to be **eventually** yield the same workflow as what we currently only have in a feature branch with PR https://github.com/miraisolutions/techguides/pull/27. Hence the command currently pulls in the deployed https://mirai-solutions.ch/techguides/shiny-ci-cd/actions/ci-cd.yml, which is still the old workflow file.

Still, I could assess this locally relying on a checkout of the techguides branch, via `usethis::use_github_action(url = "file://~/GitHubProjects/techguides/shiny-ci-cd/actions/ci-cd.yml")`

**Comment by riccardoporreca** [comment]
Re-tested `usethis::use_github_action(url = "https://mirai-solutions.ch/techguides/shiny-ci-cd/actions/ci-cd.yml")` after merging https://github.com/miraisolutions/techguides/pull/27, yielding the same results

**Review by riccardoporreca — APPROVED** [review]

**Comment by Chargothrond** [comment]
ah ok, I didn't think about what it meant too much, but basically the respective source file could also have been copy pasted over if I understand correctly (which is anyway what I did more or less).

**Pull request merged**
